To: vim_dev@googlegroups.com
Subject: Patch 7.3.439
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.439
Problem:    Compiler warnings to size casts in Perl interface.
Solution:   Use XS macros. (James McCoy)
Files:      src/if_perl.xs, src/typemap


*** ../vim-7.3.438/src/if_perl.xs	2011-09-07 18:47:19.000000000 +0200
--- src/if_perl.xs	2012-02-12 00:28:52.000000000 +0100
***************
*** 611,617 ****
      if (ptr->w_perl_private == NULL)
      {
  	ptr->w_perl_private = newSV(0);
! 	sv_setiv(ptr->w_perl_private, (IV)ptr);
      }
      else
  	SvREFCNT_inc(ptr->w_perl_private);
--- 611,617 ----
      if (ptr->w_perl_private == NULL)
      {
  	ptr->w_perl_private = newSV(0);
! 	sv_setiv(ptr->w_perl_private, PTR2IV(ptr));
      }
      else
  	SvREFCNT_inc(ptr->w_perl_private);
***************
*** 629,635 ****
      if (ptr->b_perl_private == NULL)
      {
  	ptr->b_perl_private = newSV(0);
! 	sv_setiv(ptr->b_perl_private, (IV)ptr);
      }
      else
  	SvREFCNT_inc(ptr->b_perl_private);
--- 629,635 ----
      if (ptr->b_perl_private == NULL)
      {
  	ptr->b_perl_private = newSV(0);
! 	sv_setiv(ptr->b_perl_private, PTR2IV(ptr));
      }
      else
  	SvREFCNT_inc(ptr->b_perl_private);
*** ../vim-7.3.438/src/typemap	2010-08-15 21:57:29.000000000 +0200
--- src/typemap	2012-02-12 00:28:52.000000000 +0100
***************
*** 6,12 ****
  T_VIOBJNOMUNGE
  	if (sv_isa($arg, \"${ntype}\")) {
  	    IV tmp = SvIV((SV*)SvRV($arg));
! 	    $var = ($type) tmp;
  	    if (!tmp)
  		croak(\"$ntype no longer exists\");
  	}
--- 6,12 ----
  T_VIOBJNOMUNGE
  	if (sv_isa($arg, \"${ntype}\")) {
  	    IV tmp = SvIV((SV*)SvRV($arg));
! 	    $var = INT2PTR($type, tmp);
  	    if (!tmp)
  		croak(\"$ntype no longer exists\");
  	}
*** ../vim-7.3.438/src/version.c	2012-02-12 00:18:54.000000000 +0100
--- src/version.c	2012-02-12 00:31:18.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     439,
  /**/

-- 
Microsoft is to software what McDonalds is to gourmet cooking

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
